synfig-studio/src/gui/widgets/widget_timeslider.cpp

```diff
@@ -338,9 +365,20 @@ Widget_Timeslider::on_button_press_event(GdkEventButton *event) //for clicking
 
 	if (event->button == 1) {
 		Time time = time_plot_data->get_t_from_pixel_coord(event->x);
-		time_plot_data->time_model->set_time(time);
+		time_plot_data->time_model->set_time(time);//setting the time which is in time model
 	}
 
+	double current= time_plot_data->get_double_pixel_t_coord(time_plot_data->get_t_from_pixel_coord(event->x));
+	double x0,x1,w,z0,z1,width;
+	get_bounds_rectangle_dimensions(x0,x1,w,true);
+	get_bounds_rectangle_dimensions(z0,z1,width,false);
+
+	if((current <= (x0+w))  && (current >= (x0 + w - boundary_dimension) ))
```

Comment:
`x1` ?

Reply:
oh, yes x1 makes for a smaller condition. I updated it in the other place and forgot to here, will do. 

p.s. now I get what you meant, through x1 w/width arent really ever needed